criteria/common/test/org/immutables/criteria/ReflectionTest.java

```diff
@@ -23,16 +27,27 @@ public void reflection() {
     check(evaluate(crit.age.isIn(22, 23, 24), person));
     check(!evaluate(crit.isMarried.isTrue(), person));
     check(evaluate(crit.isMarried.isFalse(), person));
-    check(evaluate(crit.isMarried.isEqualTo(false), person));
+    check(evaluate(crit.isMarried.isFalse(), person));
 
     check(evaluate(crit.age.isAtLeast(22), person));
     check(!evaluate(crit.age.isAtLeast(23), person));
-    check(evaluate(crit.lastName.isEmpty(), person));
+    check(evaluate(crit.lastName.isAbsent(), person));
     check(!evaluate(crit.lastName.isPresent(), person));
     check(evaluate(crit.lastName.isPresent(), person.withLastName("Smith")));
-    check(!evaluate(crit.lastName.isEmpty(), person.withLastName("Smith")));
+    check(!evaluate(crit.lastName.isAbsent(), person.withLastName("Smith")));
   }
 
+  @Test
+  public void debug() {
+    PersonCriteria crit = PersonCriteria.create().lastName.isAbsent()
+            .or().or() // yuck :(
```

Comment:
looks entertaining :)

Reply:
Yes this is a little ugly. 

Probably in future one can generate two classes (with and without `or()`) something like  
```java
class CriteriaWithoutOr {
}

class Criteria<T>  extends CriteriaWithoutOr<T> {
   public Criteria<T> or() {
   }
}
```

Reply:
hm, yep, interesting to think about it